### ISSUE TITLE
Core: narrowing conversions when `float` is used as type for `scalar`

### ISSUE THREAD

**Issue by GPMueller** [spirit-code/spirit#512]
On MSVC, narrowing conversions are an error, while they are a warning on e.g. gcc.
Issues arise when `float` is used as type for `scalar`, because of code such as
```C++
std::vector<Vector3> Geometry::BravaisVectorsHex2D60()
{
    return { { 0.5*std::sqrt(3), -0.5, 0 },
                { 0.5*std::sqrt(3),  0.5, 0 },
                { 0,   0,                 1 } };
}
```
or `tri_positions[i] = {basis_cell_points[tri[i]].x, basis_cell_points[tri[i]].y, 0};`.

**This needs to be fixed in order to allow Windows builds of the CUDA version of Spirit.**

_Side note: for some reason gcc also narrows the `0.5` etc. in the assignment to `Vector3` without a warning, which makes it hard to find all occurences on a Linux system._

**Comment by GPMueller** [comment]
Fixed on branch 'hotfix' with 4c20f855f1670444b51610417c207481238cc1c2, to be merged into a new release. Note that the `tri_positions` fix needs to be applied to the develop branch.

**Comment by GPMueller** [comment]
This should be fixed on master with release of v1.8.7 and on develop with 1043fa5220f2ba546403d25e96bd8acf7d67b929.
